modules/navigation/3d/godot_navigation_server_3d.cpp

```diff
@@ -1490,84 +1436,30 @@ Vector<Vector3> GodotNavigationServer3D::simplify_path(const Vector<Vector3> &p_
 
 	p_epsilon = MAX(0.0, p_epsilon);
 
-	LocalVector<uint32_t> simplified_path_indices = get_simplified_path_indices(p_path, p_epsilon);
-
-	uint32_t indices_count = simplified_path_indices.size();
-
-	Vector<Vector3> simplified_path;
-	simplified_path.resize(indices_count);
-
-	Vector3 *w = simplified_path.ptrw();
-	const Vector3 *r = p_path.ptr();
-	for (uint32_t i = 0; i < indices_count; i++) {
-		w[i] = r[simplified_path_indices[i]];
-	}
-
-	return simplified_path;
-}
-
-LocalVector<uint32_t> GodotNavigationServer3D::get_simplified_path_indices(const Vector<Vector3> &p_path, real_t p_epsilon) {
-	p_epsilon = MAX(0.0, p_epsilon);
-	real_t squared_epsilon = p_epsilon * p_epsilon;
-
-	LocalVector<bool> valid_points;
-	valid_points.resize(p_path.size());
-	for (uint32_t i = 0; i < valid_points.size(); i++) {
-		valid_points[i] = false;
-	}
-
-	simplify_path_segment(0, p_path.size() - 1, p_path, squared_epsilon, valid_points);
-
-	int valid_point_index = 0;
-
-	for (bool valid : valid_points) {
-		if (valid) {
-			valid_point_index += 1;
+	LocalVector<Vector3> source_path;
+	{
+		source_path.resize(p_path.size());
+		const Vector3 *r = p_path.ptr();
+		for (uint32_t i = 0; i < p_path.size(); i++) {
+			source_path[i] = r[i];
 		}
 	}
 
-	LocalVector<uint32_t> simplified_path_indices;
-	simplified_path_indices.resize(valid_point_index);
-	valid_point_index = 0;
+	LocalVector<uint32_t> simplified_path_indices = NavMeshQueries3D::get_simplified_path_indices(source_path, p_epsilon);
```

Comment:
Because it did not work. The compiler vomited all over the place trying to give the CoW Vector to the LocalVector parameter. Theoretically the LocalVector template should convert but it didn't. The internal simplify functions all use LocalVector to avoid the CoW farm. It is only the server api that need to stick to CoW Vector as scripts can not work with LocalVector.